File: csharp/src/DarkStrata.CredentialCheck/Exceptions/TimeoutException.cs
using System;

namespace DarkStrata.CredentialCheck;

/// <summary>
/// Exception thrown when a request times out.
/// </summary>
public sealed class DarkStrataTimeoutException : DarkStrataException
{
    /// <summary>
    /// The timeout duration in milliseconds.
    /// </summary>
    public int TimeoutMs { get; }

    /// <summary>
    /// Creates a new DarkStrataTimeoutException.
    /// </summary>
    /// <param name="timeoutMs">The timeout duration in milliseconds.</param>
    /// <param name="innerException">The inner exception (if any).</param>
    public DarkStrataTimeoutException(int timeoutMs, Exception? innerException = null)
        : base($"Request timed out after {timeoutMs}ms", ErrorCode.TimeoutError, isRetryable: true, innerException: innerException)
    {
        TimeoutMs = timeoutMs;
    }
}

File: sdks/java/src/main/java/io/darkstrata/credentialcheck/exception/TimeoutException.java
package io.darkstrata.credentialcheck.exception;

/**
 * Exception thrown when a request times out.
 */
public class TimeoutException extends DarkStrataException {

    private final long timeoutMs;

    public TimeoutException(long timeoutMs) {
        this(timeoutMs, null);
    }

    public TimeoutException(long timeoutMs, Throwable cause) {
        super("Request timed out after " + timeoutMs + "ms", ErrorCode.TIMEOUT_ERROR, null, true, cause);
        this.timeoutMs = timeoutMs;
    }

    /**
     * Get the timeout duration in milliseconds.
     */
    public long getTimeoutMs() {
        return timeoutMs;
    }
}
